opossum-tool/OpossumUI
# feat: confirm changes to previously preferred attributions immediately

### Summary of changes

- create generic confirmation dialog with custom handlers for onConfirm and onCancel
- make user confirm changes to previously preferred attributions immediately rather than on save

### Context and reason for change

Closes #2433.

### How can the changes be tested

Two ways:
1. Make changes to an existing previously-preferred attribution. Notice that you are immediately prompted to confirm your intended change, even before hitting "save". If you confirm, you will lose the previously-preferred chip before even hitting save. Once you save, you also lose the grey star on the package card.
2. Add a previously-preferred attribution via the "+" button of an autocomplete. Notice that you are NOT being asked to make any confirmations and that you are NOT losing the previously-preferred chip/star when you save the attribution. But if you try to make any changes BEFORE hitting "save" you will be asked to confirm your intended change and you will be losing the previously-preferred chip.
